src/v/resource_mgmt/storage.cc

```diff
@@ -92,13 +92,35 @@ ss::future<> disk_space_manager::stop() {
         _storage_node->local().unregister_disk_notification(
           node::disk_type::data, _data_disk_nid);
     }
+    _as.request_abort();
     _control_sem.broken();
     co_await _gate.close();
 }
 
 ss::future<> disk_space_manager::run_loop() {
     vassert(ss::this_shard_id() == run_loop_core, "Run on wrong core");
 
+    // Always wait for the trim interval after startup to ensure that
```

Comment:
for my understanding.. this is still a best effort wait, right? There maybe some odd partition (for whatever reason) hasn't updated its max_collectible_offset.

Reply:
That's right. We'd have to wait for all stms to fully catch up to guarantee that.